### PR TITLE
feature/count

### PR DESCRIPTION
**Pull request by baywet**
- codegen update: adds count odata query option
- adds base methods for odata query option count
- removes extra code gen that shouldn't have been generated
- removes the unused app root constant - adds a constant for text rutun type
- adds a method to read the odata count
- adds the ability to parse raw text responses
- adds classes for primitive response types
- code-gen update: raw count support
- fail safe in case we can't parse the long
- adds test for raw and query count

### CODE REVIEW

File: src/main/java/com/microsoft/graph/http/PrimitiveRequestBuilder.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import com.microsoft.graph.concurrency.ICallback;
+import java.util.Arrays;
+import java.util.EnumSet;
+import javax.annotation.Nullable;
+import javax.annotation.Nonnull;
+import com.microsoft.graph.core.IBaseClient;
+import com.microsoft.graph.http.BaseRequestBuilder;
+
+/**
+ * The class for the Call Record Request Builder.
```
Comment (zengin) [inline src/main/java/com/microsoft/graph/http/PrimitiveRequestBuilder.java:36]: Comment doesn't match.

File: src/main/java/com/microsoft/graph/http/PrimitiveRequestBuilder.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    public PrimitiveRequestBuilder(@Nonnull final String requestUrl, @Nonnull final IBaseClient client, @Nullable final java.util.List<? extends com.microsoft.graph.options.Option> requestOptions, @Nonnull final Class<T> primitiveTypeClass) {
+        super(requestUrl, client, requestOptions);
+        this._returnTypeClass = primitiveTypeClass;
+    }
+
+    /**
+     * Creates the request
+     *
+     * @param requestOptions the options for this request
+     * @return the CallRecordRequest instance
```
Comment (zengin) [inline src/main/java/com/microsoft/graph/http/PrimitiveRequestBuilder.java:58]: Comment doesn't match.

File: src/main/java/com/microsoft/graph/http/PrimitiveRequestBuilder.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
+// ------------------------------------------------------------------------------
+// Copyright (c) 2017 Microsoft Corporation
```
Comment (zengin) [inline src/main/java/com/microsoft/graph/http/PrimitiveRequestBuilder.java:2]:
```suggestion
// Copyright (c) 2020 Microsoft Corporation
```

File: src/main/java/com/microsoft/graph/http/PrimitiveRequestBuilder.java (hunk 1 above)
Comment (baywet) [reply src/main/java/com/microsoft/graph/http/PrimitiveRequestBuilder.java]: good catch, thanks. Fixed.

File: src/main/java/com/microsoft/graph/http/PrimitiveRequestBuilder.java (hunk 2 above)
Comment (baywet) [reply src/main/java/com/microsoft/graph/http/PrimitiveRequestBuilder.java]: good catch, thanks. Fixed.

File: src/main/java/com/microsoft/graph/http/PrimitiveRequestBuilder.java (hunk 3 above)
Comment (baywet) [reply src/main/java/com/microsoft/graph/http/PrimitiveRequestBuilder.java]: fixed manually with other comments issues

File: src/main/java/com/microsoft/graph/requests/extensions/WorkbookChartSeriesCollectionRequestBuilder.java [hunk 4]
```diff
@@ -21,7 +21,7 @@
 import com.microsoft.graph.requests.extensions.WorkbookChartSeriesItemAtRequestBuilder;
 import com.microsoft.graph.http.BaseCollectionRequestBuilder;
 import com.microsoft.graph.core.IBaseClient;
-
+import com.microsoft.graph.http.PrimitiveRequestBuilder;
```
Comment (MIchaelMainer) [inline src/main/java/com/microsoft/graph/requests/extensions/WorkbookChartSeriesCollectionRequestBuilder.java:24]: This probably shouldn't have this import as it isn't used and I don't think it makes sense here.

**Review by MIchaelMainer — APPROVED** [review]
I like the default count().

File: src/main/java/com/microsoft/graph/requests/extensions/WorkbookChartSeriesCollectionRequestBuilder.java (hunk 4 above)
Comment (MIchaelMainer) [reply src/main/java/com/microsoft/graph/requests/extensions/WorkbookChartSeriesCollectionRequestBuilder.java:24]: I don't know why it is here, although there is a count method composed

Comment (baywet) [reply src/main/java/com/microsoft/graph/requests/extensions/WorkbookChartSeriesCollectionRequestBuilder.java]: fixed. I wanted to keep the template simpler but inserting references when not needed can be confusing. Plus it has a (limited) impact on build performance for us

**Review by zengin — APPROVED** [review]

**Pull request merged**